ci/purge_expired_questionnaire_state.yaml

```diff
@@ -0,0 +1,48 @@
+platform: linux
+image_resource:
+    type: docker-image
+    source:
+        repository: gcr.io/google.com/cloudsdktool/cloud-sdk
+params:
+    SERVICE_ACCOUNT_JSON: ((gcp.service_account_json))
+    PROJECT_ID:
+    DATAFLOW_TEMPLATE_VERSION: '2021-08-02-00_RC00'
+    EXPIRATION_TIME_OFFSET_IN_SECONDS: 3600
+run:
+    path: bash
+    args:
+        - -exc
+        - |
+            export GOOGLE_APPLICATION_CREDENTIALS=/root/gcloud-service-key.json
+            cat >$GOOGLE_APPLICATION_CREDENTIALS <<EOL
+            $SERVICE_ACCOUNT_JSON
+            EOL
+
+            export CURRENT_DATE=`date --utc +%s`
+            export SECONDS=$((${CURRENT_DATE} - EXPIRATION_TIME_OFFSET_IN_SECONDS))
+            export DATETIME=`date --utc --date @${SECONDS} +"%Y-%m-%dT%H:%M:%SZ"`
+
+            gcloud auth activate-service-account --key-file "${GOOGLE_APPLICATION_CREDENTIALS}"
+            gcloud config set project "${PROJECT_ID}"
+
+            DATAFLOW_JOB=$(gcloud dataflow jobs run purge_questionnaire_state \
+            --region europe-west2 \
+            --gcs-location gs://dataflow-templates/"${DATAFLOW_TEMPLATE_VERSION}"/Datastore_to_Datastore_Delete \
+            --parameters \
+            datastoreReadGqlQuery="SELECT * FROM \`questionnaire-state\` WHERE submitted_at < DATETIME(\"${DATETIME}\")",\
```

Comment:
Just realised this isn't going to actually compare dates. Turns out, the `submitted_at` is stored as a string in Datastore. We just deserialise to a `datetime` from the `str`, so we can't properly compare date without changing this or using timestamps again.

Reply:
Regardless of timestamp/datetime, the `WHERE` clause also needs to check the submitted at is not `null`.